### PR TITLE
*: fix grpc test

### PR DESCRIPTION
**Pull request by nolouch**
Merge to https://github.com/pingcap/pd/pull/493
cc @disksing @overvenus

### CODE REVIEW

**Comment by siddontang** [comment]
This PR contains other changes, can you let `disksing/grpc` merge master first?

**Comment by nolouch** [comment]
@siddontang  this PR will merge to branch `disksing/grpc` firstly. It just let test work and resolves conflicts with the master.

**Comment by overvenus** [comment]
@siddontang @nolouch I am going to merge this branch to `disksing/grpc`, then I will do a massive clean up on the `disksing/grpc`.

File: pkg/metricutil/metricutil_test.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ import (
 	"time"
 
 	. "github.com/pingcap/check"
-	"github.com/pingcap/kvproto/pkg/pdpb"
+	//	"github.com/pingcap/kvproto/pkg/pdpb"
```
Comment (siddontang) [inline pkg/metricutil/metricutil_test.go]: remove this

File: pkg/metricutil/metricutil_test.go [hunk 2]
```diff
@@ -63,6 +63,7 @@ func (s *testMetricsSuite) TestCamelCaseToSnakeCase(c *C) {
 	}
 }
 
+/*
```
Comment (siddontang) [inline pkg/metricutil/metricutil_test.go]: use multi `//` for comment, or remove this directly.

File: server/cluster_test.go [hunk 3]
```diff
@@ -14,12 +14,16 @@
 package server
 
 import (
+	"context"
```
Comment (siddontang) [inline server/cluster_test.go]: use golang.org context

File: pkg/metricutil/metricutil_test.go (hunk 1 above)
Comment (overvenus) [inline pkg/metricutil/metricutil_test.go]: I will address your comments in `disksing/grpc`. :grin:

File: server/api/server_test.go [hunk 4]
```diff
@@ -14,6 +14,8 @@
 package api
 
 import (
+	"context"
```
Comment (siddontang) [inline server/api/server_test.go]: ditto

File: server/cluster_worker_test.go [hunk 5]
```diff
@@ -14,14 +14,14 @@
 package server
 
 import (
+	"context"
```
Comment (siddontang) [inline server/cluster_worker_test.go]: ditto

File: server/id_test.go [hunk 6]
```diff
@@ -14,7 +14,7 @@
 package server
 
 import (
-	"math/rand"
+	"context"
```
Comment (siddontang) [inline server/id_test.go]: ditto

File: server/leader_test.go [hunk 7]
```diff
@@ -14,6 +14,7 @@
 package server
 
 import (
+	"context"
```
Comment (siddontang) [inline server/leader_test.go]: ditto

File: server/tso_test.go [hunk 8]
```diff
@@ -14,76 +14,56 @@
 package server
 
 import (
-	"net"
+	"context"
```
Comment (siddontang) [inline server/tso_test.go]: ditto

File: pkg/metricutil/metricutil_test.go (hunk 1 above)
Comment (siddontang) [inline pkg/metricutil/metricutil_test.go]: No, please fix all here, then I will merge your PR.

**Pull request merged**